File: src/Common/Desnz.Chmm.ApiClients/Services/UserService.cs
using Desnz.Chmm.ApiClients.Http;
using Desnz.Chmm.Common.Dtos;
using Desnz.Chmm.Identity.Common.Commands;
using Desnz.Chmm.Identity.Common.Dtos.ManufacturerUser;
using Microsoft.AspNetCore.Http;

namespace Desnz.Chmm.ApiClients.Services
{
    public class UserService : HttpServiceClient, IUserService
    {
        /// <summary>
        /// WARNING: All Uris here MUST NOT have a leading /
        /// </summary>
        public UserService(HttpClient client, IHttpContextAccessor httpContextAccessor) : base(client, httpContextAccessor)
        {
        }

        public async Task<HttpObjectResponse<List<ChmmUserDto>>> GetAll()
        {
            return await HttpGetAsync<List<ChmmUserDto>>("api/identity/users/admin");
        }

        public async Task<HttpObjectResponse<List<ChmmUserDto>>> GetAdminUsers()
        {
            return await HttpGetAsync<List<ChmmUserDto>>("api/identity/users/admin");
        }

        public async Task<HttpObjectResponse<List<ViewManufacturerUserDto>>> GetManufacturerUsers(Guid organisationId)
        {
            return await HttpGetAsync<List<ViewManufacturerUserDto>>($"api/identity/users/manufacturer/{organisationId}");
        }

        public async Task<HttpObjectResponse<ChmmUserDto>> GetAdminUser(Guid userId)
        {
            return await HttpGetAsync<ChmmUserDto>($"api/identity/users/admin/{userId}");
        }

        public async Task<HttpObjectResponse<object>> InviteAdminUser(InviteAdminUserCommand command)
        {
            return await HttpPostAsync<object>("api/identity/users/admin", command);
        }

        public async Task<HttpObjectResponse<object>> ActivateAdminUser(ActivateAdminUserCommand command)
        {
            return await HttpPutAsync<object>("api/identity/users/admin/activate", command);
        }

        public async Task<HttpObjectResponse<object>> DeactivateAdminUser(DeactivateAdminUserCommand command)
        {
            return await HttpPutAsync<object>("api/identity/users/admin/deactivate", command);
        }

        public async Task<HttpObjectResponse<object>> UpdateUser(EditAdminUserCommand command)
        {
            return await HttpPutAsync<object>("api/identity/users/admin", command);
        }
    }
}

File: src/Web/Desnz.Chmm.Web/ClientApp/src/app/stores/account-management/services/user-service.ts
import { HttpClient, HttpEvent } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { InviteAdminUserCommand } from '../commands/invite-admin-user-command';
import { EditUserCommand } from '../commands/edit-user-command';
import { ChmmUser } from 'src/app/stores/account-management/dtos/chmm-user';
import { ActivateAdminUserCommand } from '../commands/activate-admin-user-command';
import { DeactivateAdminUserCommand } from '../commands/deactivate-admin-user-command';
import { InviteManufacturerUserCommand } from '../commands/invite-manufacturer-user-command';
import { EditManufacturerUserCommand } from '../commands/edit-manufacturer-user-command';

@Injectable()
export class UserService {
  private readonly _baseUrl: string;

  constructor(private _httpClient: HttpClient) {
    this._baseUrl = '/api/identity/users';
  }

  getAllUsers(): Observable<ChmmUser[]> {
    return this._httpClient.get<ChmmUser[]>(`${this._baseUrl}/all`, {
      responseType: 'json',
    });
  }

  getAdminUsers(): Observable<ChmmUser[]> {
    return this._httpClient.get<ChmmUser[]>(`${this._baseUrl}/admin`, {
      responseType: 'json',
    });
  }

  getManufacturerUsers(organisationId?: string | null): Observable<ChmmUser[]> {
    return this._httpClient.get<ChmmUser[]>(
      `${this._baseUrl}/manufacturer/${organisationId}`,
      {
        responseType: 'json',
      }
    );
  }

  getAdminUser(userId: string): Observable<ChmmUser> {
    return this._httpClient.get<ChmmUser>(`${this._baseUrl}/admin/${userId}`, {
      responseType: 'json',
    });
  }

  inviteAdminUser(command: InviteAdminUserCommand): Observable<void> {
    return this._httpClient.post<void>(`${this._baseUrl}/admin`, command, {
      responseType: 'json',
    });
  }

  editAdminUser(command: EditUserCommand): Observable<void> {
    return this._httpClient.put<void>(`${this._baseUrl}/admin`, command, {
      responseType: 'json',
    });
  }

  editManufacturerUser(command: EditManufacturerUserCommand | null): Observable<void> {
    return this._httpClient.put<void>(`${this._baseUrl}/manufacturer`, command, {
      responseType: 'json',
    });
  }

  activateAdminUser(command: ActivateAdminUserCommand): Observable<void> {
    return this._httpClient.put<void>(
      `${this._baseUrl}/admin/activate`,
      command,
      {
        responseType: 'json',
      }
    );
  }

  deactivateAdminUser(command: DeactivateAdminUserCommand): Observable<void> {
    return this._httpClient.put<void>(
      `${this._baseUrl}/admin/deactivate`,
      command,
      {
        responseType: 'json',
      }
    );
  }

  inviteManufacturerUser(command: InviteManufacturerUserCommand): Observable<void> {
    return this._httpClient.post<void>(`${this._baseUrl}/manufacturer`, command, {
      responseType: 'json',
    });
  }

  getManufacturerUser(organisationId: string, userId: string): Observable<ChmmUser> {
    return this._httpClient.get<ChmmUser>(`${this._baseUrl}/manufacturer/${organisationId}/users/${userId}`, {
      responseType: 'json',
    });
  }

  deactivateManufacturerUser(userId: string, organisationId: string): Observable<void> {
    return this._httpClient.post<void>(`${this._baseUrl}/manufacturer/deactivate`, { userId, organisationId }, {
      responseType: 'json',
    });
  }
}
